pkg/api/v1alpha1/kepler_types.go

```diff
@@ -78,18 +85,41 @@ type ModelServerSpec struct {
 	Path string `json:"path,omitempty"`
 
 	// +kubebuilder:default=""
-	RequiredPath string `json:"requiredPath,omitempty"`
+	RequestPath string `json:"requestPath,omitempty"`
 
 	// +kubebuilder:default=""
-	PromServer string `json:"promServer,omitempty"`
+	ListPath string `json:"listPath,omitempty"`
 
 	Trainer *ModelServerTrainerSpec `json:"trainer,omitempty"`
+
+	// +kubebuilder:default=""
+	PipelineURL string `json:"pipelineUrl,omitempty"`
+
+	// +kubebuilder:default=""
+	ErrorKey string `json:"errKey,omitempty"`
 }
 
 type EstimatorSpec struct {
+	Node      EstimatorGroup `json:"node,omitempty"`
+	Container EstimatorGroup `json:"container,omitempty"`
+}
+
+type EstimatorGroup struct {
+	Total      *EstimatorConfig `json:"total,omitempty"`
+	Components *EstimatorConfig `json:"components,omitempty"`
+}
+
+type EstimatorConfig struct {
+	// +kubebuilder:default=false
+	SidecarEnabled bool `json:"sidecar,omitempty"`
```

Comment:
isn't `Enabled` this better? We don't need to be specific about `how` a feature is enabled. That is implementation detail. 

```suggestion
	Enabled bool `json:"enabled,omitempty"`
```

Reply:
Kepler estimator could mean the local estimator and sidecar estimator. The config here (e.g., initURL) is a common for both estimator. If sidecar is not enabled, the local estimator inside kepler exporter will be used.

Reply:
How about we then specify that in the spec itself 
```
estimator: 
  enabled: true|false
  mode: sidecar|built-in 
```

Reply:
We will have only with and without sidecar. I think it is okay to have only sidecar spec to be true or false.